src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TupleNameCompletionProviderTests.cs

```diff
@@ -0,0 +1,139 @@
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider() => new TupleNameCompletionProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task AfterOpenParen()
+        {
+            await VerifyItemExistsAsync(@"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            (int word, int zword) t = ($$
+        }
+    }
+}", "word:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task AfterOpenParenWithBraceCompletion()
+        {
+            await VerifyItemExistsAsync(@"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            (int word, int zword) t = ($$)
+        }
+    }
+}", "word:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task AfterComma()
+        {
+            await VerifyItemExistsAsync(@"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            (int word, int zword) t = (1, $$
+        }
+    }
+}", "zword:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task AfterCommaWithBraceCompletion()
+        {
+            await VerifyItemExistsAsync(@"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            (int word, int zword) t = (1, $$)
+        }
+    }
+}", "zword:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InTupleAsArgument()
+        {
+            await VerifyItemExistsAsync(@"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main((int word, int zword) args)
+        {
+             Main(($$))
+        }
+    }
+}", "word:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task MultiplePossibleTuples()
+        {
+            var markup = @"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main((int number, int znumber) args) { }
+        static void Main((string word, int zword) args) {
+            Main(($$
+        }
+    }
+}";
+            await VerifyItemExistsAsync(markup, "word:");
+            await VerifyItemExistsAsync(markup, "number:");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task MultiplePossibleTuplesAfterComma()
+        {
+            var markup = @"
+namespace ConsoleApp36
+{
+    class Program
+    {
+        static void Main((int number, int znumber) args) { }
+        static void Main((string word, int zword) args) {
+            Main(($$
+        }
+    }
+}";
+            await VerifyItemExistsAsync(markup, "zword:");
+            await VerifyItemExistsAsync(markup, "znumber:");
```

Comment:
Do we have tests that validate that this works property when part of the item has already been typed?  Also, what if we have:

```(a$$, b)```